src/ast.c

```diff
@@ -161,6 +161,17 @@ static value_t fl_defined_julia_global(fl_context_t *fl_ctx, value_t *args, uint
     return (b != NULL && jl_atomic_load_relaxed(&b->owner) == b) ? fl_ctx->T : fl_ctx->F;
 }
 
+static value_t fl_nothrow_julia_global(fl_context_t *fl_ctx, value_t *args, uint32_t nargs)
+{
+    // tells whether a var is defined, in the sense that accessing it is nothrow
+    argcount(fl_ctx, "nothrow-julia-global", nargs, 1);
+    (void)tosymbol(fl_ctx, args[0], "nothrow-julia-global");
+    jl_ast_context_t *ctx = jl_ast_ctx(fl_ctx);
+    jl_sym_t *var = jl_symbol(symbol_name(fl_ctx, args[0]));
+    jl_binding_t *b = jl_get_module_binding(ctx->module, var, 0);
```

Comment:
Doesn't that cause the binding to resolve? Currently merely lowering a function that mentions a symbol does not resolve that binding in case you want to define it to something else later.

Reply:
alright, `jl_get_binding_if_bound` then